### PR TITLE
[FIX] stock: create new quality checks to pass for backorders

### PR DESCRIPTION
**Pull request by len-foss**
There was no Quality check entry for backorders
when the user did a partial shipment.
We simply confirm the newly created backorder picking which creates
the relevant quality checks if needed.

opw-1816845

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -813,31 +813,26 @@ def _get_overprocessed_stock_moves(self):
...
-                'backorder_id': picking.id
-            })
-            picking.message_post(body=_("Back order <em>%s</em> <b>created</b>.") % (backorder_picking.name))
-            not_done_bo_moves.write({'picking_id': backorder_picking.id})
-            if not picking.date_done:
-                picking.write({'date_done': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT)})
-            backorder_picking.action_confirm()
-            backorder_picking.action_assign()
-            backorders |= backorder_picking
+            bo_moves_todo = backorder_moves.filtered(lambda move: move.state not in ('done', 'cancel'))
```
Comment (nim-odoo) [inline addons/stock/models/stock_picking.py]: Why not simply copy-paste what was in `_action_done`?

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -813,31 +813,26 @@ def _get_overprocessed_stock_moves(self):
...
+                    'name': '/',
+                    'move_lines': [],
+                    'move_line_ids': [],
+                    'backorder_id': picking.id
+                })
+                picking.message_post(body=_("Back order <em>%s</em> <b>created</b>.") % (backorder_picking.name))
+                bo_moves_todo.write({'picking_id': backorder_picking.id})
+                if not picking.date_done:
+                    picking.write({'date_done': time.strftime(DEFAULT_SERVER_DATETIME_FORMAT)})
+                backorder_picking.action_confirm()
```
Comment (nim-odoo) [inline addons/stock/models/stock_picking.py]: I think it was agreed to not do this... Is there a reason to add it?

File: addons/stock/models/stock_picking.py [hunk 3]
```diff
@@ -813,31 +813,26 @@ def _get_overprocessed_stock_moves(self):
 
     @api.multi
     def _create_backorder(self, backorder_moves=[]):
-        """ Move all non-done lines into a new backorder picking. If the key 'do_only_split' is given in the context, then move all lines not in context.get('split', []) instead of all non-done lines.
+        """ Move all non-done lines into a new backorder picking.
         """
-        # TDE note: o2o conversion, todo multi
         backorders = self.env['stock.picking']
         for picking in self:
             backorder_moves = backorder_moves or picking.move_lines
```
Comment (nim-odoo) [inline addons/stock/models/stock_picking.py]: I have the feeling that `backorder_moves` has become useless here, especially if it is used inside a loop on the recordset.

File: addons/stock/models/stock_picking.py [hunk 4]
```diff
@@ -813,31 +813,26 @@ def _get_overprocessed_stock_moves(self):
...
-            backorders |= backorder_picking
+            bo_moves_todo = backorder_moves.filtered(lambda move: move.state not in ('done', 'cancel'))
+            if bo_moves_todo:
+                backorder_picking = picking.copy({
+                    'name': '/',
+                    'move_lines': [],
+                    'move_line_ids': [],
+                    'backorder_id': picking.id
+                })
+                picking.message_post(body=_("Back order <em>%s</em> <b>created</b>.") % (backorder_picking.name))
```
Comment (nim-odoo) [inline addons/stock/models/stock_picking.py]: Why take the "unused" message? Once again, is there a reason to not copy-paste what was on `_action_done`?

File: addons/stock/models/stock_move.py [hunk 5]
```diff
@@ -983,7 +983,7 @@ def _create_extra_move(self):
                 precision_rounding=self.product_uom.rounding,
                 rounding_method ='UP')
             extra_move_vals = self._prepare_extra_move_vals(extra_move_quantity)
-            extra_move = self.copy(default=extra_move_vals)._action_confirm()
+            extra_move = self.with_context(skip_check=True).copy(default=extra_move_vals)._action_confirm()
```
Comment (nim-odoo) [inline addons/stock/models/stock_move.py]: This should be in a separate commit with an explanation of the use case.

File: addons/stock/models/stock_picking.py (hunk 4 above)
Comment (len-odoo) [reply addons/stock/models/stock_picking.py]: I think you're right, I change that immediately.

File: addons/stock/models/stock_move.py (hunk 5 above)
Comment (len-odoo) [reply addons/stock/models/stock_move.py]: All right.

File: addons/stock/models/stock_picking.py [hunk 6]
```diff
@@ -812,32 +812,26 @@ def _get_overprocessed_stock_moves(self):
         )
 
     @api.multi
-    def _create_backorder(self, backorder_moves=[]):
```
Comment (nim-odoo) [inline addons/stock/models/stock_picking.py]: I know it's stupid, but you should keep it in the method parameters, to avodi breaking overrides. However, I don't see the point of using it in the method itself. Probably to remove in master branch
